### PR TITLE
DCOS-11750: add size to external volumes

### PR DESCRIPTION
**Pull request by Poltergeist**
This introduces size to external volumes.
Fixes volumes on root level in the reducer.
And also fixes container type to be set with external volumes.

### CODE REVIEW

**Comment by nLight** [comment]
:bug: External volume doesn't parse size back from JSON

**Comment by Poltergeist** [comment]
Should work now.

File: plugins/services/src/js/components/forms/VolumesFormSection.js [hunk 1]
```diff
@@ -165,13 +165,19 @@ class VolumesFormSection extends Component {
...
-    return data.map((volumes, key) => {
+    return data.map((volume, key) => {
       const nameError = errorsLens
         .at(key + offset, {})
         .attr('external', {})
         .get(this.props.errors)
         .name;
 
+      const sizeError = errorsLens
+      .at(key, {})
```
Comment (nLight) [inline plugins/services/src/js/components/forms/VolumesFormSection.js]: our new `indent` eslint rule still doesn't catch that, we should consider `newline-per-chained-call` but would you please add some spaces?

File: plugins/services/src/js/reducers/serviceForm/ExternalVolumes.js [hunk 2]
```diff
@@ -119,6 +127,9 @@ module.exports = {
       if (type === SET && `externalVolumes.${index}.containerPath` === joinedPath) {
         state[index].containerPath = value;
       }
+      if (type === SET && `externalVolumes.${index}.size` === joinedPath) {
+        state[index].size= value;
```
Comment (nLight) [inline plugins/services/src/js/reducers/serviceForm/ExternalVolumes.js]: adding new eslint rule on my list, would you add missing space here and 3 lines below?

**Comment by wavesoft** [comment]
Functionality is there! We just need to rebase the whole branch I guess in order to make sure that the eslint rules will pass?

**Comment by Poltergeist** [comment]
Addressed your comments and rebased @nLight

**Comment by nLight** [comment]
💪  merge at will

**Comment by MatApple** [comment]
run integration tests

**Comment by mlunoe** [comment]
Rebased to get integration tests passing. Will  as soon as testes () pass

**Comment by mlunoe** [comment]
run integration tests

**Pull request merged**